### DISCUSSION TITLE
Access URL from File Manager to use in html editor?

### DISCUSSION THREAD

**Discussion by ctag** [GetPublii/Publii#1830] [Help] [answered]
Hi,

I'd like to embed a .webm video in a post. To do that I created a text link to the file in the File Manager, and then used the source editor to convert it to a video tag. 

```
<p><video width="400" height="150" loop="loop">
	<source src="#INTERNAL_LINK#/file/media/files/example.webm" type="video/webm" /></video></p>
<!-- Original link from WYSIWYG <p><a href="#INTERNAL_LINK#/file/media/files/example.webm">Image</a></p> -->
```

But when previewing the post, #INTERNAL_LINK# is not replaced, and the video element is blank.

I think I'm missing something obvious. I'd appreciate any pointers!

**Comment by bobmitro** [comment, marked as answer]
It will work very well, but the path you’re using is incorrect. Publii allows you to place files either in the root directory or in the media/files directory. So, move the `.webm` file to the `/media/files/ `folder in the media manager. Then, in the embed code, use `#INTERNAL_LINK#/file/example.webm`.

And now it should work fine, although please note that the movie won't be visible in the post preview; you will need to preview the entire page to see it.

**Reply by ctag** [reply]
Thank you for the explanation! 
Previewing the full site or syncing to the real site works, but previewing the single post in a browser does not replace the URL.

**Comment by Vinc26** [comment]
Hi,

In my case, it doesn't work. I'm using a HTML block to add script (slideshow), with a code like this:

```
<div id='slider' class='slider'></div>
<script  type='text/javascript'>
    var containerId = 'slider';

    var options = {
        bullets: false,
        arrows:false,
		auto: true,
		autoTime: 1000,
		transitionTime: 3000,
    	autoPauseOnHover: false,
        images: [
            '#INTERNAL_LINK#/file/Aurelie-Lamour-Photographe-Blanc.jpg',
            '#INTERNAL_LINK#/file/Aurelie-Lamour-Photographe-Portrait-Studio-Enfant-Ancrage.jpg',
            '#INTERNAL_LINK#/file/Aurelie-Lamour-Photographe-Packshot-Studio-Bois-Joss-Naigeon.jpg',
        .........
```

🤷‍♂️ When previewing or viewing, `#INTERNAL_LINK#/file/` si not replaced by the correct relative path. It's not replaced at all. Why ?

**Reply by dziudek** [reply]
@Vinc26 - your internal links are not working, because our current replacement code expects `"` at the end - please use `"` instead of `'` in `images` array and it will work properly :) 

I will add a fix in v.0.46.3 which will allow also single quotes in internal links handling :)

**Reply by candidexmedia** [reply]
@dziudek : Is there any way to get internal links (ex: `#INTERNAL_LINK#/page/18`) working with the Theme and Post/Page Settings API?

Trying to add them causes rendering issues: https://github.com/GetPublii/Publii/discussions/2306#discussioncomment-14318985

**Reply by dziudek** [reply]
@candidexmedia - I have added this issue to checklist for work on the next release

**Reply by candidexmedia** [reply]
Great, thank you!
